mongo_device_dao.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+type mongoDeviceDAO struct {
+	devices *mongo.Collection
+}
+
+func (dao *mongoDeviceDAO) Save(device Device) (Device, error) {
+	countDocuments, err := dao.devices.CountDocuments(context.Background(), bson.M{})
```

Comment:
context should be passed. you wanna take it from request in handler (`r.Context()`) and then pass it down to every layer. Mongo might need it to handle e.g. request cancelation.

You have to get use to starting most of your functions with `func funcName(ctx context.Context, ...)` 😄 

Same goes for rest of methods too.